services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/modules/db/repositories/osparc.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+import sqlalchemy as sa
+from models_library.projects import ProjectID
+from simcore_postgres_database.models.projects import projects
+from simcore_postgres_database.models.users import users
+
+from ._base import BaseRepository
+
+_logger = logging.getLogger(__name__)
+
+
+class OSparcRepository(BaseRepository):
+    async def get_user_email(self, user_id: int) -> str | None:
+        async with self.db_engine.connect() as conn:
+            result = await conn.execute(
+                sa.select(users.c.email).where(users.c.id == user_id)
+            )
+            data = result.first()
```

Comment:
dont' you need to await this?  https://aiopg.readthedocs.io/en/stable/sa.html#aiopg.sa.ResultProxy.first

Reply:
No this is fine :+1: btw i am using AsyncEngine here